stdlib/3/smtplib.pyi

```diff
@@ -39,71 +44,77 @@ class SMTPConnectError(SMTPResponseException): ...
-    def connect(self, host=..., port=..., source_address=...): ...
-    def send(self, s): ...
-    def putcmd(self, cmd, args=...): ...
+    sock = ...  # type: Optional[socket]
+    def connect(self, host: str = ..., port: int = ...,
+                source_address: Optional[_SourceAddress] = ...) -> _Reply: ...
+    def send(self, s: Union[bytes, str]) -> None: ...
+    def putcmd(self, cmd: str, args: str = ...) -> None: ...
     def getreply(self) -> _Reply: ...
-    def docmd(self, cmd, args=...): ...
-    def helo(self, name=...): ...
-    def ehlo(self, name=...): ...
-    def has_extn(self, opt): ...
-    def help(self, args=...): ...
+    def docmd(self, cmd: str, args: str = ...) -> _Reply: ...
+    def helo(self, name: str = ...) -> _Reply: ...
+    def ehlo(self, name: str = ...) -> _Reply: ...
+    def has_extn(self, opt: str) -> bool: ...
+    def help(self, args: str = ...) -> bytes: ...
     def rset(self) -> _Reply: ...
     def noop(self) -> _Reply: ...
     def mail(self, sender: str, options: Sequence[str] = ...) -> _Reply: ...
     def rcpt(self, recip: str, options: Sequence[str] = ...) -> _Reply: ...
-    def data(self, msg): ...
-    def verify(self, address): ...
-    vrfy = ...  # type: Any
-    def expn(self, address): ...
-    def ehlo_or_helo_if_needed(self): ...
-    def login(self, user, password): ...
-    def starttls(self, keyfile=..., certfile=..., context=...): ...
+    def data(self, msg: Union[bytes, str]) -> _Reply: ...
+    def verify(self, address: str) -> _Reply: ...
+    vrfy = verify
+    def expn(self, address: str) -> _Reply: ...
+    def ehlo_or_helo_if_needed(self) -> None: ...
+    if sys.version_info >= (3, 5):
+        def login(self, user: str, password: str, *,
+                  initial_response_ok: bool = ...) -> _Reply: ...
+    else:
+        def login(self, user: str, password: str) -> _Reply: ...
+    def starttls(self, keyfile: Optional[str] = ..., certfile: Optional[str] = ...,
+                 context: Optional[SSLContext] = ...) -> _Reply: ...
     def sendmail(self, from_addr: str, to_addrs: Union[str, Sequence[str]],
                  msg: Union[bytes, str], mail_options: Sequence[str] = ...,
                  rcpt_options: List[str] = ...) -> _SendErrs: ...
     def send_message(self, msg: _Message, from_addr: Optional[str] = ...,
                      to_addrs: Optional[Union[str, Sequence[str]]] = ...,
                      mail_options: List[str] = ...,
                      rcpt_options: Sequence[str] = ...) -> _SendErrs: ...
-    def close(self): ...
+    def close(self) -> None: ...
     def quit(self) -> _Reply: ...
 
 class SMTP_SSL(SMTP):
-    default_port = ...  # type: Any
-    keyfile = ...  # type: Any
-    certfile = ...  # type: Any
-    context = ...  # type: Any
-    def __init__(self, host=..., port=..., local_hostname=..., keyfile=..., certfile=...,
-                 timeout=..., source_address=..., context=...): ...
+    keyfile = ...  # type: Optional[str]
```

Comment:
We prefer variable annotations (`x: int`) in new code.

Reply:
Hm, good to know, but this wasn't really new code -- the particular highlighted line was changed from `keyfile = ...  # type: Any` to `keyfile = ...  # type: Optional[str]`, and similar changes were made to `certfile` and `context`.

There are quite a few variables annotated using the older formatting in smtplib.pyi, would you like me to change all of them to use variable annotations instead?

Reply:
Yes, let's just change them all. Thanks!